components/mdns/mdns.c

```diff
@@ -5346,7 +5351,7 @@ static void _mdns_service_task(void *pvParameters)
     for (;;) {
         if (_mdns_server && _mdns_server->action_queue) {
             if (xQueueReceive(_mdns_server->action_queue, &a, portMAX_DELAY) == pdTRUE) {
-                if (a && a->type == ACTION_TASK_STOP) {
+                if (a == NULL || a->type == ACTION_TASK_STOP) {
```

Comment:
Question: does `a == NULL` mean that an exception occurs? If yes, at this time, should we stop mdns timer? Should we free the lock and deinit the queue? How about asserting right now?

Reply:
It's definitely an invalid state, which should never happen -- i.e. typically happens during mem corrupt issues.
`assert()` is probably a better reaction then just stopping the task.
Have fixed, thanks for the suggestion!